js/src/Ice/ConnectionI.js

```diff
@@ -1154,6 +1176,10 @@ export class ConnectionI {
                 let message = this._sendStreams.shift();
                 this._writeStream.swap(message.stream);
                 message.sent();
+                if (message.receivedReply) {
```

Comment:
It would be helpful to add a comment explaining how message.receivedReply can be true.
There is nothing obviously async in this code. Where do we "suspend" to give a chance to receive the reply?

Reply:
I will add a comment.

Here when we call sendNextMessage the first message is always already sent. If the message was sent asynchronously,  and the reply is received almost immediately the Red ready with the reply might be processed before than the Write ready that ends up calling sendNextMessage.